### DISCUSSION TITLE
Curious the use of Vue vs something like SvelteKit

### DISCUSSION THREAD

**Discussion by maietta** [louislam/dockge#163] [General]
Don't get me wrong, I love Vue. But I left Vue last year for SvleteKit.

I see that you have a separate backend and front-end. Was there a specific reason for keeping the backend and front-end code separate?

What I really enjoy about SvelteKit is that my server side and client side code live in the same project space. I feel like this provides some key advantages over other systems.

I hope I am not sounding rude, as people love the frameworks they use. TypeScript and Vue are good. I'm curious about people's thoughts on this.

**Comment by louislam** [comment]
The main reason is just my preference, because it is the only modern reactive framework that I have learnt. Also there were a lot of things and code from my another project Uptime Kuma that I could reuse, so that I could kick-start very very fast. Indeed I just used around 3 weeks to build Dockge 1.0.0.

> Was there a specific reason for keeping the backend and front-end code separate?

I think we have no choice with Vue?

But thanks for introducing SvelteKit, I always thought SvelteKit is just another SPA framework and I never get into it. Didn't know it also doing backend things. It is interesting. I am planning to create Dockge's simple website. I think it is a chance to try SvelteKit.

**Reply by maietta** [reply]
Thank you for your answer.

Would you be open to this project being ported to SvelteKit if someone like me offered to take on the challenge?

I understand you are already building fast right now, so maybe this stage of build would not be best choice if so.

Or perhaps I could maintain a fork, but doing so feels like I would be taking away from your thunder ⛈️ 

It's a great project. Would love to see it streamlined with SvelteKit.

**Reply by louislam** [reply]
Dockge is under MIT license, as long as you dont remove my copyright info, you can do whatever you want.

But I would suggest that you should come up with a new idea and create your own project rather than reinventing the wheel.

I can share ideas here if you would like to go this path.

**Reply by maietta** [reply]
My needs for a SvelteKit version of this software would be so that I can leverage some of the core functions within my own software stack I've built for my startup that would only ever be used in-house.

I just figured that if I was going to go through the trouble of doing this, I may as well offer to contribute to the project in a way that could benefit everyone and not just me. It was merely a tentative offer in case SvelteKit could have been appealing over Vue + TS.

I think forking the software just to port to SvelteKit as an alternative to yours, feature by feature, function by function, would be pointless on it's face and that feels a lot like copying and taking away from you. I don't like that idea and would not entertain that idea. I know how I would feel if someone copied my projects like that.

Anyway, I won't fork the project for the sake of porting to SvelteKit. Instead, I will use your project for inspiration for my own work.

Thank you for the conversation.
